src/admin/redux/modules/email/reducer.js

```diff
@@ -81,8 +82,14 @@ const template_subs = ifElse
 
 const indexByProp = compose(indexBy, propOr(''))
 
+const greeting = member => assoc
+  ( 'greeting'
+  , member.first_name || member.title + ' ' + member.last_name
+  , member
+  )
+
 const [ primaries, secondaries ] =
-  map(indexByProp, [ 'primary_email', 'secondary_email' ])
+  map(compose(map(map(greeting)), indexByProp), [ 'primary_email', 'secondary_email' ])
```

Comment:
Ok PR is open, we will continue working on redux/modules/letters.js (see
commit message)

On Mon, Jun 20, 2016 at 2:04 PM, Jack Murphy <jmurphy.web@gmail.com> wrote:

> We should get a pull request open by 3.
>
>
> On Fri, Jun 17, 2016 at 3:15 PM, Jack Murphy <jmurphy.web@gmail.com>
> wrote:
>
>> These sound like some delightfully probing questions. Will take a look.
>>
>> On Fri, Jun 17, 2016 at 12:46 PM, Rafe <notifications@github.com> wrote:
>>
>>> In src/admin/redux/modules/email/reducer.js
>>> <https://github.com/foundersandcoders/sail-back/pull/261#discussion_r67498020>
>>> :
>>>
>>> >  const [ primaries, secondaries ] =
>>> > -  map(indexByProp, [ 'primary_email', 'secondary_email' ])
>>> > +  map(compose(map(map(greeting)), indexByProp), [ 'primary_email', 'secondary_email' ])
>>>
>>> This line could maybe be rewritten. See if you can understand it though.
>>>
>>> Hint: the inner map and the outer map are mapping over different functor
>>> instances. Do you know what the first one is? Do you know the second?
>>>
>>> —
>>> You are receiving this because you were assigned.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/foundersandcoders/sail-back/pull/261/files/25d17365e130fd1b63bf63f944e0f9ca33341f1f#r67498020>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe/AMDYNc-JBlVc-Yy4Y04divJKYhoQzEuKks5qMokBgaJpZM4I4SCB>
>>> .
>>>
>>
>>
>